### PR TITLE
27 change product type method

### PR DESCRIPTION
**Pull request by lojzatran**
Fixes: https://github.com/commercetools/commercetools-node-variant-reassignment/issues/27

### CODE REVIEW

File: lib/utils/utils.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export function unpublish (version) {
+  return {
+    version,
+    actions: [{ action: 'unpublish' }]
+  }
+}
+
+export function publish (version) {
```
Comment (junajan) [inline lib/utils/utils.js]: This `utils/utils/publish` function is not used anywhere in the code

File: spec/unit/product-manager.spec.js [hunk 2]
```diff
@@ -130,4 +134,46 @@ describe('ProductManager', () => {
       ])
     })
   })
+
+  describe('compare two products', () => {
+    it('should equal when product ids are equal', () => {
+      productService = new ProductManager(utils.logger, {})
```
Comment (junajan) [inline spec/unit/product-manager.spec.js]: This init of productService can be done in `beforeEach`?

File: lib/utils/utils.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+export function unpublish (version) {
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply lib/utils/utils.js]: https://github.com/commercetools/commercetools-node-variant-reassignment/pull/30/commits/7016cd557776114b58a5bc364ae2285bfb80f5a1

File: lib/utils/utils.js (hunk 1 above)
Comment (lojzatran) [reply lib/utils/utils.js]: https://github.com/commercetools/commercetools-node-variant-reassignment/pull/30/commits/7016cd557776114b58a5bc364ae2285bfb80f5a1

File: lib/services/transaction-manager.js [hunk 4]
```diff
@@ -20,9 +20,13 @@ export default class TransactionManager {
   }
 
   createTransaction (transaction, productId) {
+    return this.createOrUpdateTransactionValue(transaction, this._getTransactionKey(productId))
+  }
+
+  createOrUpdateTransactionValue (transaction, key) {
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply lib/services/transaction-manager.js]: https://github.com/commercetools/commercetools-node-variant-reassignment/pull/30/commits/5fd4834c7737f3b63d40266bf7f1bcfbe839fddb

File: spec/unit/product-manager.spec.js (hunk 2 above)
Comment (lojzatran) [reply spec/unit/product-manager.spec.js]: https://github.com/commercetools/commercetools-node-variant-reassignment/pull/30/commits/74990e7b1ab9e4d53c0d8574ef129156ce447a3b

File: lib/services/transaction-manager.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (lojzatran) [reply lib/services/transaction-manager.js]: https://github.com/commercetools/commercetools-node-variant-reassignment/pull/30/commits/5fd4834c7737f3b63d40266bf7f1bcfbe839fddb

File: lib/services/product-manager.js [hunk 5]
```diff
@@ -270,4 +271,35 @@ export default class ProductManager {
 
     return Boolean(projectionPublished || masterDataPublished)
   }
+
+  async changeProductType (product, newProductTypeId) {
+    if (product.masterData.published)
+      await this.client.products
+        .byId(product.id)
+        .update(unpublish(product.version))
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply lib/services/product-manager.js]: https://github.com/commercetools/commercetools-node-variant-reassignment/pull/30/commits/7016cd557776114b58a5bc364ae2285bfb80f5a1#diff-15ce2e3c2d77d574a8c8ae56d19b9d92R377

**Comment by junajan** [comment]
Approved now ... could you please fix failing test and then it can be merged.

**Pull request merged**